CMakeLists.txt

```diff
@@ -99,14 +99,19 @@ else()
   message(STATUS "Not using VT insertable collections")
 endif()
 
-find_package(Kokkos 4.0 REQUIRED)
+find_package(Kokkos 4.1 REQUIRED)
 
 if (Kokkos_CXX_STANDARD LESS 17)
   message(FATAL_ERROR "distBVH requires Kokkos built with C++17 or later")
 endif()
 set_property(TARGET bvh PROPERTY CXX_STANDARD ${Kokkos_CXX_STANDARD})
 target_link_libraries(bvh PUBLIC Kokkos::kokkos)
 
+option(BVH_ENABLE_CUDA "Enable CUDA" OFF)
+if (BVH_ENABLE_CUDA)
+  target_compile_definitions(bvh PUBLIC BVH_ENABLE_CUDA)
+endif()
```

Comment:
I think here we should use `Kokkos_ENABLE_CUDA` instead of a dedicated option here and maybe write out a message to specify it's using CUDA (we can still keep the BVH_ENABLE_CUDA define). That way we won't run into the situation where Kokkos is not built with CUDA enabled but BVH is

Reply:
Switched to `Kokkos_ENABLE_CUDA`.

Since Kokkos is now a hard requirement, I have removed a couple of occurences of `BVH_ENABLE_CUDA` in the code too. We are now left with only one (where `primary_execution_space` is decided):
```
distBVH  git grep bvh_enable_cuda
CMakeLists.txt:112:  target_compile_definitions(bvh PUBLIC BVH_ENABLE_CUDA)
src/bvh/util/kokkos.hpp:46:#ifdef BVH_ENABLE_CUDA
```

Reply:
Note that not having a dedicated option means that CUDA will be used by default as `primary_execution_space` whenever it is available. This might break users' applications until GPU support is ready (#16).